modules/internal/DefaultAssociative.chpl

```diff
@@ -496,8 +496,13 @@ module DefaultAssociative {
               }
             }
           }
+          when ArrayInit.gpuInit {
+            // may not be too difficult, not a priority at the moment
+            halt("Associative arrays cannot be initialized on GPU locales with",
```

Comment:
Can you add this to the limitations section on the tech note:

https://chapel-lang.org/docs/technotes/gpu.html#requirements-and-limitations

Reply:
I am not wild about that. We don't list every limitation we have (no distributed arrays for example), and this is way more niche than that. To me, that list is intentionally non-comprehensive.

Also, the error message is self-explanatory (as compared to distributed arrays are crashing at the moment, say).

Do you want to argue strongly for including this here? I don't have any other arguments against that I am hiding :)

Reply:
I kind of do want to argue for that.

I really want to be able to tell people "with GPU support you can use all the normal Chapel features on GPUs except for things we document as "known limitations" on the tech note.

Sure, realistically we'll have unknown limitations that (by definition) we've overlooked. But when we encounter a limitation I think we should document it there. If the list becomes overwhelming we can try and think about how to better sort or organize it. But I think it improves our credibility and sends a powerful message if we can say "we handle everything except for these publicly listed things here."

As far as distributed arrays. You're right thats a big oversight missing from that list. We should add it. Anyone trying to determine if Chapel's GPU support will work for them or not would want to know about that.

Reply:
Convincing.

> But I think it improves our credibility and sends a powerful message if we can say "we handle everything except for these publicly listed things here.

Especially this part is appealing. OK will do.

Reply:
Thanks!

As an aside: it's probably worth thinking about how what we've done for 1.30 effects this list (and if we're missing anything else big or obvious, not sure how I didn't think of "distributed arrays" before).

As a TODO item for me: I'll update it for where we are with AMD support.

Reply:
Hah! I beat you to it. https://github.com/chapel-lang/chapel/pull/21753#discussion_r1127180716